rust/gateway/src/eventloop.rs

```diff
@@ -156,22 +162,35 @@ impl Eventloop {
                 }) => {
                     let tunnel = Arc::clone(&self.tunnel);
 
-                    match self.connection_request_tasks.try_push(
-                        (req.client.id, req.reference.clone()),
-                        async move {
-                            let conn = tunnel
-                                .set_peer_connection_request(
-                                    req.client.payload,
-                                    req.client.peer.into(),
-                                    req.relays,
-                                    req.client.id,
-                                    req.expires_at,
-                                    req.resource,
-                                )
-                                .await?;
-                            Ok(GatewayResponse::ConnectionAccepted(conn))
-                        },
-                    ) {
+                    let connection_request = async move {
+                        let local_params = tunnel
+                            .set_peer_connection_request(
+                                req.client.payload.ice_parameters,
+                                req.client.peer.into(),
+                                req.relays,
+                                req.client.id,
+                            )
+                            .await?;
+
+                        let resource = resolve_resource_description(req.resource).await?;
+
+                        let domain_response = tunnel.allow_access(
+                            resource,
+                            req.client.id,
+                            req.expires_at,
+                            req.client.payload.domain,
+                        );
```

Comment:
This happens here to ensure it is still atomic from the portal's perspective, i.e. we return the `domain_response` as part of the `ConnectionAccepted`. Long term, I think that should be separate and the portal should send us a separate `allow_access` instruction.